gossip/service.go

```diff
@@ -78,6 +89,77 @@ func (f *ServiceFeed) SubscribeNewLogs(ch chan<- []*types.Log) notify.Subscripti
 	return f.scope.Track(f.newLogs.Subscribe(ch))
 }
 
+func (f *ServiceFeed) Start(store *evmstore.Store) {
```

Comment:
I think this new functionality should also be covered through unit tests. A basic coverage of properties like the in-order notification of blocks would be great.

By abstracting the `*evmstore.Store`, building tests for this should also be rather straight forward, without the need of a lot of setup code; I have an example [here](https://github.com/0xsoniclabs/sonic/commit/d5e2c876905fa00e408820fefc442492ece31e23).